src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -104,33 +105,33 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
+                                projectModel.AddProjectInformation(message: string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoPackagesFoundForFrameworks, projectModel.ProjectName), problemType: ProblemType.Information);
                             }
                             else
                             {
                                 if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
                                 {
                                     PopulateSourceRepositoryCache(listPackageArgs);
-                                    WarnForHttpSources(listPackageArgs);
+                                    WarnForHttpSources(listPackageArgs, projectModel);
                                     await GetRegistrationMetadataAsync(packages, listPackageArgs);
                                     await AddLatestVersionsAsync(packages, listPackageArgs);
                                 }
 
-                                bool printPackages = FilterPackages(packages, listPackageArgs);
+                                bool printPackages = projectModel.PrintPackagesFlag;
 
                                 // Filter packages for dedicated reports, inform user if none
                                 if (listPackageArgs.ReportType != ReportType.Default && !printPackages)
                                 {
                                     switch (listPackageArgs.ReportType)
                                     {
                                         case ReportType.Outdated:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectName));
+                                            projectModel.AddProjectInformation(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoUpdatesForProject, projectModel.ProjectName), ProblemType.Information);
                                             break;
                                         case ReportType.Deprecated:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoDeprecatedPackagesForProject, projectName));
+                                            projectModel.AddProjectInformation(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoDeprecatedPackagesForProject, projectModel.ProjectName), ProblemType.Information);
                                             break;
                                         case ReportType.Vulnerable:
-                                            Console.WriteLine(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoVulnerablePackagesForProject, projectName));
+                                            projectModel.AddProjectInformation(string.Format(CultureInfo.CurrentCulture, Strings.ListPkg_NoVulnerablePackagesForProject, projectModel.ProjectName), ProblemType.Information);
```

Comment:
I feel like this is more of a renderer/outputter concern than a data model issue. It's been a long time since I've looked at the json output spec, but do we even want any of these messages for json?  If there are no updates, deprecated packages or vulnerabilities, then the appropriate json array will have zero elements. I'm not sure if these messages will be output in the json, but if it is, then it feels redundant to me. 🤷

Reply:
`ProblemType.Information` type project information is ignored by json output, but it's used by console renderer.
Could we leave it as it's for now? Because in order to move this logic into console renderer we need to pass around `printPackages` flag inside our projectModel.